test/shared/src/main/scala-3/zio/test/CompileVariants.scala

```diff
@@ -44,6 +44,9 @@ trait CompileVariants {
   inline def assertTrue(inline exprs: => Boolean*): TestResult =
     ${SmartAssertMacros.smartAssert('exprs)}
 
+  inline def assertTrue(inline expr: Any, inline exprs: => Any*): UIO[Nothing] =
```

Comment:
Could the issue be that `assertTrue` doesn't have an overload that takes the first param without `*`? does adding this fixes the proble?

```scala 3
inline def assertTrue(inline expr: Boolean, inline exprs: Boolean*): TestResult
```

Also, I don't think we need the arguments to be by name when they're inlined (inline arguments are by nature "by-name"). I think we can reduce the generated codesize by removing all the `=>` from the methods

Reply:
I realized the issue is simply that `implicit def lift` for test result is defined in a `TestResult` object, instead of the `zio.test` package object.